testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,165 @@
+      response_type: "TestSessionResults"
+      metadata_type: "TestSessionMetadata"
+    };
+  }
+}
+
+message Container {
+  enum Kind {
+    // No kind was specified, so the service will attempt to infer one.
+    //
+    // This inference works as follows:
+    //   1. If this field belongs to the driver_container of the
+    //      StartTestSessionRequest, this kind is inferred to be a driver.
+    //   2. Otherwise, this field is inferred to be a client worker.
+    UNSPECIFIED = 0;
+
+    // Orchestrates the test amongst workers and reports results.
+    DRIVER = 1;
+
+    // Processes incoming requests from client containers.
+    SERVER = 2;
+
+    // Sends outgoing requests to a server container.
+    CLIENT = 3;
+  }
+
+  // The name of the image as it appears in the registry.
+  //
+  // When using GCR, this will likely be a fully-qualified URL. For example,
+  // the name of the Java worker may be "gcr.io/grpc-testing/java_qps_worker".
+  string name = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The kind which assigns the responsibilities of this container.
+  Kind kind = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  oneof version_identifier {
+    // The hash assigned to the image.
+    string sha256 = 2;
+
+    // A tag which was explicitly set on the image.
+    //
+    // For example, a developer may explicitly tag an image for repetitive
+    // testing and sharing. They may add a tag with a PR number or version.
+    //
+    // Note: the "latest" tag is automatically set by the registry, and there
+    // are no guarantees that it will point to a specific image. It should be
+    // avoided.
+    string tag = 3;
+  }
+
+  // The number of pods that should be created with this container. For example,
+  // specify 2 replicas on a client container to create 2 clients.
+  //
+  // This field will always be ignored on driver containers. It is ignored on
+  // server containers at present, but may be supported eventually.
+  int32 replicas = 4;
+}
+
+message StartTestSessionRequest {
+  // The test scenarios to run as a part of the test session.
+  repeated grpc.testing.Scenario scenarios = 1;  // [(google.api.field_behavior) = REQUIRED];
+
+  // The container used as the driver which runs the test scenarios.
+  //
+  // If unset, the "latest" tag is used with the standard driver.
+  Container driver_container = 2;  // [(google.api.field_behavior) = OPTIONAL];
+
+  // The containers used as clients.
+  repeated Container worker_containers = 3;  // [(google.api.field_behavior) = REQUIRED];
+
+  // An optional location where the results should be written.
+  oneof result_destination {
+    // The name of the BigQuery table where results should be written.
+    string bq_results_table = 4;  // [(google.api.field_behavior) = OPTIONAL];
+
+    // The fully qualified GCS URI where a results file should be written.
+    string gcs_results_file_uri = 5;  // [(google.api.field_behavior) = OPTIONAL];
```

Comment:
This field may not be supported in the first version of the prototype. This is designed to offer a way to access the raw JSON results file without going through BigQuery. It may be unnecessary. Let me know your thoughts.